### PR TITLE
feat(community): Add AirtableLoader to load documents from Airtable with retry and pagination handling

### PR DESCRIPTION
**Pull request by SkSirius**
As per discussion #7064

**Summary:**

This PR introduces the AirtableLoader class, a document loader for fetching records from Airtable using the LangChain framework. The loader supports features such as pagination, retry logic, and environment-based API token validation to facilitate reliable document loading from Airtable tables.

**Changes:**

Introduced AirtableLoader class that supports synchronous and asynchronous generator-based document loading.
- load(): Retrieves all documents from Airtable with automatic pagination.
- loadLazy(): An asynchronous generator that yields each document individually to manage memory effectively.

**Testing:**
- Unit and integration tests included

### CODE REVIEW

File: libs/langchain-community/src/document_loaders/web/airtable.ts [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    url: string,
+    attempt = 1
+  ): Promise<AirtableResponse> {
+    try {
+      return await this.fetchRecords(url);
+    } catch (error) {
+      const maxRetries = AirtableLoader.MAX_RETRIES;
+      const retryDelayMs = AirtableLoader.DEFAULT_RETRY_DELAY_MS;
+
+      if (attempt <= maxRetries) {
```
Comment (jacoblee93) [inline libs/langchain-community/src/document_loaders/web/airtable.ts:131]: Generally prefer the builtin `AsyncCaller` for consistency:

https://github.com/langchain-ai/langchainjs/blob/main/langchain-core/src/utils/async_caller.ts

How to use (it's a built-in field on the base `Embeddings` class): https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-openai/src/chat_models.ts#L1821

**Comment by jacoblee93** [comment]
Thanks! See small comment

File: libs/langchain-community/src/document_loaders/web/airtable.ts (hunk 1 above)
Comment (jacoblee93) [reply libs/langchain-community/src/document_loaders/web/airtable.ts:131]: Import example

https://github.com/langchain-ai/langchainjs/blob/e54c10124d793614031664367622271da8e35114/langchain/src/util/prompt-layer.ts#L2

**Comment by SkSirius** [comment]
@jacoblee93, updated loader to us AsyncCaller.

File: libs/langchain-community/src/document_loaders/web/airtable.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    this.baseId = baseId;
+    this.kwargs = kwargs;
+
+    if (!this.apiToken) {
+      throw new Error(
+        "Missing Airtable API token. Please set AIRTABLE_API_TOKEN environment variable."
+      );
+    }
+
+    this.asyncCaller = new AsyncCaller({ maxRetries: 3, maxConcurrency: 5 });
```
Comment (jacoblee93) [inline libs/langchain-community/src/document_loaders/web/airtable.ts:58]: May be nice to make this configurable in the future but fine for now

**Review by jacoblee93 — APPROVED** [review]
Thanks!

**Pull request merged**